eldritch/items/spells.py

```diff
@@ -170,6 +170,20 @@ def get_usable_interrupt(self, event, owner, state):
     )
     return events.CastSpell(owner, self, choice=choice)
 
+  def get_interrupt(self, event, owner, state):
+    if (
+        isinstance(event, (events.DiscardSpecific, events.DiscardNamed))
+        and event.character == owner
+        and event.discarded
+        and self in event.discarded
```

Comment:
For DiscardNamed, the `event.discarded` will be the individual item, so this `in` check will fail. This is probably on me, as I never should have made those two different events, but we should probably work around it for now.